### PR TITLE
Two improvements for toggle switch 

### PR DESCRIPTION
**Pull request by solikhver**
I have made two improvements that could be useful for others:
1. Add inner color builder instead of color value. In this case inner color changed from one to another using linear interpolation which looks better than solid color that we theoretically can change when value changed.
2. Use exceptionally steep calculation for opacity of indicator during transition from one step to another.

There are also some minor changes like dependencies update and example update to show new adjustments.

### CODE REVIEW

File: lib/src/widgets/animated_toggle_switch.dart [hunk 1]
```diff
@@ -513,7 +522,8 @@ class AnimatedToggleSwitch<T> extends StatelessWidget {
           (global.textDirection == TextDirection.rtl ? -1.0 : 1.0);
       final pos = global.position;
       int first = pos.floor();
-      double transitionValue = pos - first;
+      double transitionValue =
+          Interval(0, 1, curve: Curves.easeInOutExpo).transform(pos - first);
```
Comment (MaddinMade) [inline lib/src/widgets/animated_toggle_switch.dart:526]: Should we make the curve customizable via the constructors at this point or would that be too complex for users?

**Comment by MaddinMade** [comment]
@solikhver Thank you for your pull request. I'll take a closer look at it. I hope it wasn't too hard to read into my code.

**Comment by MaddinMade** [comment]
https://user-images.githubusercontent.com/43761463/193145377-63fc897c-061b-454d-b941-8c736fb51389.mov

Not only the color animation, but all transition animations have been adapted to this new curve. However, this means that the rolling animation no longer looks natural.

**Comment by solikhver** [comment]
You are right. It should be separate transition value for rotation and for opacity. Sorry for incomplete solution.   I will adjust it asap.

**Comment by MaddinMade** [comment]
@solikhver Are my changes acceptable to you?

**Comment by solikhver** [comment]
Yes, looks good. 
Out of the scope. I would propose to adjust custom constructor to have possibility to create fully customisable dual. I am talking about this._iconArrangement = IconArrangement.row; which can't be adjusted. I propose to make _iconArrangement not private and optional with default value IconArrangement.row. 
What do you think?

**Comment by MaddinMade** [comment]
This would probably cause some constructors to no longer work as desired. If you want to make such changes, you should use `CustomAnimatedToggleSwitch` directly.
With `AnimatedToggleSwitch.dual` `IconArrangement.row` for example makes no sense.
Do you have an explicit use case where you want to use `AnimatedToggleSwitch.dual` with `IconArrangement.row`?

**Comment by solikhver** [comment]
Let me a little bit explain. For example I have some special case where I am using `AnimatedToggleSwitch` as a dual toggle. Finally I see that default constructor  `dual` has some restrictions that I can't overcome. So in that case I would try to use another constructor that gives me more freedom. Of course I would go and try `custom` constructor. But in this case I will also stumble upon restriction that I can't specify `_iconArrangement` and giving up. 

Direct usage of `CustomAnimatedToggleSwitch` means that the end user have to copy your `AnimatedToggleSwitch` completely and modify it. It is that last resort that user can solve also by making fork instead of copying full content of `AnimatedToggleSwitch`. What actually I did in the end.

Hope I explained it clearly. 

P.S. just realized that it was not clear for me as a new user which approach is suitable for me for achieving my goals. Is it enough `dual`/`custom` constructors or should I use `CustomAnimatedToggleSwitch`? I have spent my time to try all of them.

**Comment by MaddinMade** [comment]
What kind of switch are you trying to achieve exactly and what kind of restrictions do you have with `dual()` exactly? Just the `IconArrangement`? I've always used either `rolling()` or `dual()` for my "dual switches" and had everything I needed.
I understand that the package is relatively clunky, but that's probably more of a documentation issue.
`AnimatedToggleSwitch.custom()` hasn't been revised for a long time either and I only didn't remove it because I don't want to create big breaking changes. This constructor was originally what `CustomAnimatedToggleSwitch` is now.

**Comment by solikhver** [comment]
Talking about my special case I was not happy with indicator animation. Not all icons looks good when they constantly rolling and gradually disappearing. 

Whole this discussion is just a food for thought. I don't want to break any code of users, but it would be nice to extend library and make it more convenient. So I am voting for constructor that allows to customize parameters, even `_iconArrangement`.

**Comment by MaddinMade** [comment]
Okay, I will make the `IconArrangement` public in `custom()`. But not for the other constructors for now.

**Comment by solikhver** [comment]
Thank you for your work and constructive communication! 
Hope that new changes are only for the best.

**Comment by MaddinMade** [comment]
@solikhver I forgot all about this pull request. I'm sorry. If the effort is not too big, you could create a new pull request with just your changes to the dart code (without dependency updates) in one commit. I like to separate features and dependency updates.

Then I can merge and publish this pull request.

**Pull request closed without merging**